xdsl/transforms/experimental/lower_hls.py

```diff
@@ -0,0 +1,202 @@
+from dataclasses import dataclass
+
+from xdsl.pattern_rewriter import (
+    PatternRewriter,
+    PatternRewriteWalker,
+    RewritePattern,
+    GreedyRewritePatternApplier,
+    op_type_rewrite_pattern,
+)
+from xdsl.ir import Block, MLContext, Region, Operation, OpResult
+from xdsl.irdl import VarOperand, VarOpResult
+from xdsl.dialects.func import FuncOp, Return
+from xdsl.dialects.func import Call
+from xdsl.dialects import builtin
+from xdsl.dialects.builtin import i32, IndexType
+from xdsl.dialects.arith import Constant
+
+from xdsl.dialects.experimental.hls import PragmaPipeline, PragmaUnroll, PragmaDataflow
+
+from xdsl.passes import ModulePass
+
+from xdsl.dialects.scf import ParallelOp, For, Yield
+
+from typing import cast, Any
+
+
+@dataclass
+class PragmaPipelineToFunc(RewritePattern):
+    def __init__(self, op: builtin.ModuleOp):
+        self.module = op
+
+    @op_type_rewrite_pattern
+    def match_and_rewrite(self, op: PragmaPipeline, rewriter: PatternRewriter, /):
+        # TODO: can we retrieve data directly without having to go through IntegerAttr -> IntAttr?
```

Comment:
Yes but I'm not sure how long merging it will take :) The current pattern is to check `isinstance(op.ii, OpResult) and isinstance(op.ii.op, MyOp)` and then going from there